tests/components/weatherflow/conftest.py

```diff
@@ -41,20 +41,30 @@ def mock_config_entry2() -> MockConfigEntry:
 def mock_has_devices() -> Generator[AsyncMock, None, None]:
     """Return a mock has_devices function."""
     with patch(
-        "homeassistant.components.weatherflow.config_flow._async_can_discover_devices",
+        "pyweatherflowudp.client.WeatherFlowListener.on",
         return_value=True,
     ) as mock_has_devices:
         yield mock_has_devices
 
 
 @pytest.fixture
-def mock_has_no_devices() -> Generator[AsyncMock, None, None]:
-    """Return a mock has_devices function returning False."""
+def mock_on_throws_timeout() -> Generator[AsyncMock, None, None]:
+    """Throw a timeout exceptoin."""
     with patch(
-        "homeassistant.components.weatherflow.config_flow._async_can_discover_devices",
-        return_value=False,
-    ) as mock_has_devices:
-        yield mock_has_devices
+        "pyweatherflowudp.client.WeatherFlowListener.on",
```

Comment:
Patch the `WeatherFlowListener` where its imported so you make sure you only patch the expected one

`homeassistant.components.weatherflow.config_flow.WeatherFlowListener.on`